nabs/preprocessors.py

```diff
@@ -197,16 +199,32 @@ def daq_step_scan_decorator(plan):
     """
 
     @wraps(plan)
-    def inner(*args, **kwargs):
-        events = kwargs.pop('events', None)
-        duration = kwargs.pop('duration', None)
-        record = kwargs.pop('record', True)
-        use_l3t = kwargs.pop('use_l3t', False)
+    def inner(*args, events=None, duration=None, record=True, use_l3t=False):
         return (yield from daq_step_scan_wrapper(plan(*args, **kwargs),
```

Comment:
I'm looking into this a bit and still trying to make it work with the `bluesky-queueserver` (honestly might have issues because i don't actually run daq - will let you know more tomorrow)
For now though it is complaining about not knowing where `**kwargs` came from here :D  - probably don't need it anymore here?

Reply:
Cool, feel free to push to this branch if you have a fix. (If not, it looks like we should just retain the old "`**kwargs`" and `kwargs.pop()`" style but add the signature-modifying stuff for now.)

I'm still not entirely convinced this is the _easiest_ way, but at least it does seem to work for fixing up the signature.

Reply:
I honestly have tried a few things without success haha - I think your solution here is great! 
And both of these worked for me:

```
def inner(*args, **kwargs):
        events = kwargs.pop('events', None)
        duration = kwargs.pop('duration', None)
        record = kwargs.pop('record', True)
        use_l3t = kwargs.pop('use_l3t', False)

and the rest of your fix here... 
```

or just add the `**kwargs` in your current one:

```
def inner(*args, events=None, duration=None, record=True, use_l3t=False, **kwargs):

and the rest of your fix here...
```

Reply:
Cool, thanks for giving this a go. I cleaned it up a bit; let's see if this still works.
As a "utility" in the new commit it now feels _less_ of a hack, kinda-sorta...